commands/GitHub_Pattern_To_Repo_Export.php

```diff
@@ -0,0 +1,226 @@
+	 *
+	 * @var string|null
+	 */
+	protected ?string $category_slug = null;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function configure() {
+		$this
+			->setDescription( 'Exports a block pattern from a site to a GitHub.' )
+			->setHelp( 'This command exports a specified block pattern into a category within a GitHub repository.' )
+			->addArgument( 'site', InputArgument::REQUIRED, 'ID or URL of the Pressable site to run the command on.' )
+			->addArgument( 'pattern-name', InputArgument::REQUIRED, 'The unique identifier of the block pattern to export (e.g., "namespace/pattern-name").' )
+			->addArgument( 'category-slug', InputArgument::REQUIRED, 'The slug of the category under which the pattern should be exported. It should be lowercase with hyphens instead of spaces (e.g., "featured-patterns").' );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function initialize( InputInterface $input, OutputInterface $output ): void {
+
+		// Retrieve the given site.
+		$this->pressable_site = get_pressable_site_input( $input, $output, fn() => $this->prompt_site_input( $input, $output ) );
+		$output->writeln( "<info>Site {$this->pressable_site->id}: {$this->pressable_site->url}</info>", Output::VERBOSITY_VERBOSE );
+
+		// Store the ID of the site in the argument field.
+		$input->setArgument( 'site', $this->pressable_site->id );
+
+		// Check if the pattern name was already provided as an argument. If not, prompt the user for it.
+		$this->pattern_name = get_string_input( $input, $output, 'pattern-name', fn() => $this->prompt_pattern_name_input( $input, $output ));
+		$output->writeln( "<info>Pattern name: {$this->pattern_name}</info>", Output::VERBOSITY_VERBOSE );
+
+		// Check if the category slug was already provided as an argument. If not, prompt the user for it.
+		$this->category_slug = slugify( get_string_input( $input, $output, 'category-slug', fn() => $this->prompt_category_slug_input( $input, $output ) ) );
+		$output->writeln( "<info>Category slug: {$this->category_slug}</info>", Output::VERBOSITY_VERBOSE );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function execute( InputInterface $input, OutputInterface $output ): int {
+
+		$output->writeln( "<fg=magenta;options=bold>Exporting {$this->pattern_name} (Category: {$this->category_slug}) from {$this->pressable_site->displayName} (ID {$this->pressable_site->id}, URL {$this->pressable_site->url})</>" );
+
+		// Upload script.
+		$sftp_connection   = \Pressable_Connection_Helper::get_sftp_connection( $this->pressable_site->id );
+		if ( \is_null( $sftp_connection ) ) {
+			$output->writeln( '<error>Could not open SFTP connection.</error>' );
+			return Command::FAILURE;
+		}
+		$result = $sftp_connection->put( '/htdocs/pattern-extract.php', file_get_contents( __DIR__ . '/../scaffold/pattern-extract.php' ) );
+		if ( ! $result ) {
+			$output->writeln( "<error>Failed to copy pattern-extract.php to {$this->pressable_site->id}.</error>" );
```

Comment:
We could do fewer checks here actually. Since the CLI is running on PHP8+, we don't have to check for `is_null( $sftp_connection )`. We can just:

```
$result = $sftp_connection?->put( ... );
if ( empty( $result ) ) {
   // return
}
```

That `?` will make sure that `put` only gets called if `$sftp_connection` is not `null`. If it is, then `$result` will contain `null`. So we can get away with just one check.

Reply:
Ditto down below with `$ssh_connection?->exec`.

Reply:
> If this fails, we can't really do the export. Shouldn't we return a Command::FAILURE?

Yes. Added that in 381d6c0

> That ? will make sure that put only gets called if $sftp_connection is not null. If it is, then $result will contain null. So we can get away with just one check.

I guess this depends on how much information we want to give when something goes wrong. With the separate checks, you'll know whether you are having a problem connecting to the site, or if there's a problem with actually pushing the file onto the server/running the command.

If you want to keep it standardized and do the single check I'll make the update, but maybe we should keep it that way for granularity when something goes wrong?

Reply:
We should probably move the check for the connection inside the Connection_Helper class actually. That way, we always get the info in the terminal.

But we can leave things as-is for now.

Reply:
Looking [further down](https://github.com/a8cteam51/team51-cli/blob/381d6c0253a65dece626229998db07b24efa0b71/commands/GitHub_Pattern_To_Repo_Export.php#L105-L106) I think this is a carry-over when I copy/pasted the code. The file isn't currently deleted from the server because the line is commented out:

```php
		// Delete script.
		// $ssh_connection->exec( 'rm /htdocs/pattern-extract.php' );
```

So if the put fails, it's because the file is already there. This made sense to save time while building it but I think that's not a great assumption to make now, so I return a failure in  381d6c0 and uncommented the `rm` command in 508f656

Reply:
> We should probably move the check for the connection inside the Connection_Helper class actually. That way, we always get the info in the terminal.

That's a good idea. We could consider returning details, as well, if it would be helpful to give that to the calling function.